cpp/oneapi/dal/algo/pca/backend/cpu/finalize_train_kernel_cov.cpp

```diff
@@ -113,17 +115,69 @@ static train_result<Task> call_daal_kernel_finalize_train(const context_cpu& ctx
 
         interop::status_to_exception(status);
     }
-
+    model_t model;
     if (desc.get_result_options().test(result_options::eigenvectors)) {
-        const auto mdl = model_t{}.set_eigenvectors(
-            homogen_table::wrap(arr_eigvec, component_count, column_count));
-        result.set_model(mdl);
+        model.set_eigenvectors(homogen_table::wrap(arr_eigvec, component_count, column_count));
     }
 
     if (desc.get_result_options().test(result_options::eigenvalues)) {
         result.set_eigenvalues(homogen_table::wrap(arr_eigval, 1, component_count));
+        model.set_eigenvalues(homogen_table::wrap(arr_eigval, 1, component_count));
+    }
+    if (desc.get_result_options().test(result_options::singular_values)) {
```

Comment:
Good point, but unfortunately current DAAL interfaces don't support variances and means as partial results in SVD, and for finalize in Cov we do Covariance: FinalizeCompute, so it could require some changes on Cov side as well. Will be aligned in separate pr